docroot/modules/custom/dct_newsletter/src/Form/NewsletterForm.php

```diff
@@ -0,0 +1,176 @@
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class NewsletterForm extends FormBase {
+
+  /**
+   * The state service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The email validator service.
+   *
+   * @var \Egulias\EmailValidator\EmailValidatorInterface
+   */
+  protected $emailValidator;
+
+  /**
+   * Constructs a NewsletterForm object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Egulias\EmailValidator\EmailValidatorInterface $email_validator
+   *   The email validator.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, EmailValidatorInterface $email_validator) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->emailValidator = $email_validator;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('email.validator')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'dct_newsletter_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+
+    $form['email'] = [
+      '#type' => 'email',
+      '#placeholder' => $this->t('Email'),
+      '#maxlength' => 60,
+      '#required' => TRUE,
+    ];
+
+    $form['error'] = [
+      '#type' => 'container',
+      '#prefix' => '<div class="newsletter-error-container">',
+      '#suffix' => '</div>',
+    ];
+
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Subscribe'),
+      '#button_type' => 'primary',
+      '#ajax' => [
+        'callback' => [$this, 'saveSubscription'],
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Submits the form using ajax.
+   *
+   * @param array $form
+   *   The subscription form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Storage for the state of the form.
+   * @return \Drupal\Core\Ajax\AjaxResponse $response
+   */
+  public function saveSubscription(array &$form, FormStateInterface $form_state) {
+    $response = new AjaxResponse();
+    $error = FALSE;
+
+    if (empty($form_state->getValue('email'))) {
+      $html = [
+        '#prefix' => '<span class="error-message">',
+        '#markup' => $this->t('This email field is required.'),
+        '#suffix' => '</span>',
+      ];
+      $error = TRUE;
+    }
+    else {
+      if (!$this->emailValidator->isValid($form_state->getValue('email'))) {
+        $html = [
+          '#prefix' => '<span class="error-message">',
+          '#markup' => $this->t('This email address is not valid.'),
+          '#suffix' => '</span>',
+        ];
+        $error = TRUE;
+      }
+
+      else {
+        $user_storage = $this->entityTypeManager->getStorage('user');
```

Comment:
You can check if the email exists, loading the user by properties. 
$users = \Drupal::entityTypeManager()->getStorage('user')->loadByProperties(['mail' => $form_state->getValue('email')]);